src/game/Maps/SpawnGroupDefines.h

```diff
@@ -78,6 +89,53 @@ struct SpawnGroupEntry
     std::vector<SpawnGroupRandomEntry*> EquallyChanced;
     std::vector<SpawnGroupRandomEntry*> ExplicitlyChanced;
     std::vector<uint32> LinkedGroups;
+
+    // may be nullptr
+    FormationEntrySPtr formationEntry;
+
+    int32 GetFormationSlotId(uint32 dbGuid) const
+    {
+        auto const& itr = std::find_if(DbGuids.begin(), DbGuids.end(), [dbGuid](SpawnGroupDbGuids const& x) { return x.DbGuid == dbGuid; });
+        return itr != DbGuids.end() ? (*itr).SlotId : -1;
+    }
+};
+
+// Formation defines
+enum SpawnGroupFormationType : uint32
+{
+    SPAWN_GROUP_FORMATION_TYPE_RANDOM              = 0,
+    SPAWN_GROUP_FORMATION_TYPE_SINGLE_FILE         = 1,
+    SPAWN_GROUP_FORMATION_TYPE_SIDE_BY_SIDE        = 2,
+    SPAWN_GROUP_FORMATION_TYPE_LIKE_GEESE          = 3,
+    SPAWN_GROUP_FORMATION_TYPE_FANNED_OUT_BEHIND   = 4,
+    SPAWN_GROUP_FORMATION_TYPE_FANNED_OUT_IN_FRONT = 5,
+    SPAWN_GROUP_FORMATION_TYPE_CIRCLE_THE_LEADER   = 6,
+
+    SPAWN_GROUP_FORMATION_TYPE_COUNT               = 7
```

Comment:
The 8th type missing here maybe? Even adding a placeholder would be cool even if its not implemented.

Reply:
what is it?